### DISCUSSION TITLE
Use camelCase in enums inside messages

### DISCUSSION THREAD

**Discussion by netanel-utila** [stephenh/ts-proto#671] [Q&A] [answered]
Given the following code:

```
message User {
  enum Status {
    STATUS_UNSPECIFIED = 0;
    STATUS_PENDING = 1;
    STATUS_ACTIVE = 2;
  }
}
```

It'll generate the following TS:
```ts
export const User_Status = { STATUS_UNSPECIFIED: 0, STATUS_PENDING: 1, STATUS_ACTIVE: 2 } as const;
```

Is there anyway to convert it to `UserStatus`?

**Comment by stephenh** [comment, marked as answer]
No, because of the concern with naming conflicts, i.e. if there was a message called `message UserStatus`.

Granted, this is unlikely, but in the huge/wide array of schemas that people use ts-proto for, "anything can happen".

We could potentially try and probe "does message UserStatus" actually exist, and if it does not, then do not use the `_` delimited name...

If you wanted to try a PR that did that, that'd be great.

**Reply by netanel-utila** [reply]
Thanks for the quick reply. If you can direct me to the correct location, I can take a look at this.
